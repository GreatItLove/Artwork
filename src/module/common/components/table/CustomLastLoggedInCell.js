import React from 'react';
import PropTypes from 'prop-types';

import convertTimestamp from '../../../../utility/convertTimestamp';

const propTypes = {
  value: PropTypes.string.isRequired
};

/**
 * @function CustomLastLoggedInCell
 * @param {Number} value
 * @returns {XML}
 */
function CustomLastLoggedInCell({ value }) {
  return <span>{value && convertTimestamp(value)}</span>;
}

CustomLastLoggedInCell.propTypes = propTypes;
export default CustomLastLoggedInCell;
