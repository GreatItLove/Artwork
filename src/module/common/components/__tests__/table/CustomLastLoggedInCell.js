import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import CustomLastLoggedInCell from '../../table/CustomLastLoggedInCell';

describe('Custom Last LoggedIn Cell', () => {
  it('Snapshot of CustomLastLoggedInCell with value', () => {
    const wrapper = shallow(<CustomLastLoggedInCell value="4654646" />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('Snapshot of CustomLastLoggedInCell without value', () => {
    const wrapper = shallow(<CustomLastLoggedInCell value=" " />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
