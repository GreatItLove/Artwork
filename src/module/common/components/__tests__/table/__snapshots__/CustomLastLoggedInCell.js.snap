// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Last LoggedIn Cell Snapshot of CustomLastLoggedInCell with value 1`] = `
<span>
  February 24, 1970 2:27 AM
</span>
`;

exports[`Custom Last LoggedIn Cell Snapshot of CustomLastLoggedInCell without value 1`] = `
<span>
  January 01, 1970 5:30 AM
</span>
`;
